### ISSUE TITLE
Protocol TypedDict Unpack not assignable with extra parameters that have defaults

### ISSUE THREAD

**Issue by esoma** [microsoft/pyright#10966]
My expectation is that `type[MyType]` is compatible with both protocols below, but `MyProtocolUsingTypedDict` detects an incompatibility with `not_required_kw`. `MyProtocolWithoutTypedDict` is compatible. I suppose there could be a subtly to `Unpack` that I'm not aware of, but my understanding is that both of these protocols should be functionally equivalent. mypy does not detect an error.

```python
from typing import Protocol, TypedDict, Unpack

class MyTypedDict(TypedDict):
    required_kw: str

class MyProtocolUsingTypedDict(Protocol):
    def __call__(self, **kwargs: Unpack[MyTypedDict]) -> object: ...

class MyProtocolWithoutTypedDict(Protocol):
    def __call__(self, *, required_kw: str) -> object: ...

class MyType:
    def __init__(self, *, not_required_kw: int = 0, required_kw: str): ...

def my_function_using_typed_dict(my_protocol: MyProtocolUsingTypedDict) -> None: ...
def my_function_without_typed_dict(my_protocol: MyProtocolWithoutTypedDict) -> None: ...

my_function_using_typed_dict(MyType)
my_function_without_typed_dict(MyType)
```

```
Argument of type "type[MyType]" cannot be assigned to parameter "my_protocol" of type "MyProtocolUsingTypedDict" in function "my_function_using_typed_dict"
  Type "type[MyType]" is not assignable to type "(**kwargs: **MyTypedDict) -> object"
    Parameter 1: type "*MyTypedDict" is incompatible with type "int"
      "*MyTypedDict" is not assignable to "int"
    Parameter "**kwargs" has no corresponding parameter  (reportArgumentType)
```

**Comment by erictraut** [comment]
By default, a `TypedDict` is an "open" structural type, which means that a compatible value may have additional keys that are not specified in the `TypedDict`. In your example, a value compatible with `MyTypedDict` must have a key named `required_kw` with a value that is assignable to type `str`. It can also contain additional keys, for example one named `foo`.

The `MyProtocolWithoutTypedDict` is not equivalent to `MyProtocolUsingTypedDict`. To make it equivalent, you'd need to redefine it as:

```python
class MyProtocolWithoutTypedDict(Protocol):
    def __call__(self, *, required_kw: str, **kwargs: object) -> object: ...
```

[PEP 728](https://peps.python.org/pep-0728/) introduces the concept of a "closed" `TypedDict`. To use this, you'll need to import `TypedDict` from `typing_extensions` (or wait for Python 3.14). Using this capability, `MyTypedDict` can be defined as "closed".

```python
class MyTypedDict(TypedDict, closed=True):
    required_kw: str
```

Closing the issue because Pyright is working correctly here and in compliance with the typing spec.

**Comment by esoma** [comment]
Got it. For anyone else encountering this issue my actual problem was closer to this not being compatible:
```python
class MyType:
    def __init__(self, *, not_required: int = 0, **kwargs: Unpack[MyTypedDict]): ...
```

I actually don't want my typed dicts to be closed, so using `extra_items=Any` fixes it. It's a bit surprising to me that it is `**kwargs: object` instead of `**kwargs: Any` by default.
